src/app/components/JumpTo/index.tsx

```diff
@@ -43,7 +43,15 @@ const JumpTo = ({ jumpToHeadings }: JumpToProps) => {
   };
 
   const titleId = 'jump-to-heading';
-
+  if (
+    showRelatedContentLink &&
+    !jumpToHeadings?.some(({ heading }) => heading === relatedContent)
```

Comment:
Yea it may be good to try and construct an array of 'things to render', including the related content if applicable, so something like:
```
  const headingsToRender = [
    // Article subheadings
    ...jumpToHeadings.map(({ heading }) => ({
      heading,
      id: idSanitiser(heading),
    })),
    // Related content
    showRelatedContentLink && {
      heading: relatedContent,
      id: 'section-label-heading-related-content-heading',
    },
  ];
```

That'll take a shallow copy of the `jumpToHeadings`, so shouldn't cause the original data to be mutated and keep appending items to it on re-renders.